src/components/common/Dialogs/CreateDomainDialog/CreateDomainDialog.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+// import { Id } from '@colony/colony-js';
+import { string, object, boolean, InferType } from 'yup';
+import { defineMessages } from 'react-intl';
+
+import Dialog, { DialogProps, ActionDialogProps } from '~shared/Dialog';
+import { ActionHookForm as Form } from '~shared/Fields';
+
+import { DomainColor } from '~types';
+import { ActionTypes } from '~redux/index';
+import { WizardDialogType, useEnabledExtensions } from '~hooks';
+import { pipe, withMeta, mapPayload } from '~utils/actions';
+
+import CreateDomainDialogForm from './CreateDomainDialogForm';
+import { getCreateDomainDialogPayload } from './helpers';
+
+type Props = DialogProps & WizardDialogType<object> & ActionDialogProps;
+
+const displayName = 'common.CreateDomainDialog';
+
+const MSG = defineMessages({
+  requiredFieldError: {
+    id: `${displayName}.requiredFieldError`,
+    defaultMessage: 'Please enter a value',
+  },
+});
+
+const validationSchema = object()
+  .shape({
+    forceAction: boolean().defined(),
+    teamName: string()
+      .max(20)
+      .required(() => MSG.requiredFieldError),
+    domainColor: string().notRequired(),
+    domainPurpose: string().max(90).notRequired(),
+    annotationMessage: string().max(4000).notRequired(),
+  })
+  .defined();
+
+type FormValues = InferType<typeof validationSchema>;
+
+const CreateDomainDialog = ({
+  callStep,
+  prevStep,
+  cancel,
+  close,
+  colony,
+}: Props) => {
+  const [isForce, setIsForce] = useState(false);
+  const navigate = useNavigate();
+
+  const { isVotingReputationEnabled } = useEnabledExtensions(colony);
```

Comment:
It's possible because we could use that hook in the `NewActionButton` and pass the data from there to the dialog wizard, however, we have to take into consideration that we also need to pass `isOneTxPaymentEnabled` and `votingReputationVersion`. So, we would need to pass 3 new props or 1 with the 3 props inside it (something like `extensionData`?).

What do you think?

Reply:
I don't have a preference, whatever feels right for you. You could possibly introduce a type that we'd use for both the return value of `useEnabledExtensions` and the 1 props you mentioned